### ISSUE TITLE
Add an optimization to remove no-op queries

### ISSUE THREAD

**Issue by brynrhodes** [cqframework/clinical_quality_language#178]
Add a feature to the translator to remove no-op queries from the output ELM. For example:

    define "Current Diagnosis Atrial FibrillationFlutter": 
      "Encounter with Principle Diagnosis of Ischemic Stroke" NonElectiveEncounter 
        where NonElectiveEncounter.diagnosis in "Atrial Fibrillation/Flutter" CurrentAtrialFibrillationFlutter

Note the trailing alias `CurrentAtrialFibrillationFlutter`. Since the query has no additional clauses, the result of the query is the same as the result of the primary source `"Atrial Fibrillation/Flutter"`. Although this is conceptually valid, there are downstream consequences in this case in that the ELM outputs with an `In` instead of an `InValueSet` because the translator doesn't detect that the right operand to the `in` is a reference to a value set.

This optimization would correct that issue, but would also be a general enhancement to the resulting ELM, simplifying where possible.

**Comment by cmoesel** [comment]
We should investigate the impact on annotations, in that there would be a more significant mismatch between the source CQL and the ELM.  For example, would it make sense to still tie the annotation for the now-optimized element to the full original source in the CQL?

**Comment by brynrhodes** [comment]
Agreed, my thinking here was that when an optimization removes a node, it replaces the source references in the remaining node with the source references in the removed node. This _should_ result in the rewrite referencing the full source CQL.
